src/Type/Row.purs

```diff
@@ -0,0 +1,27 @@
+module Type.Row
+  ( class RowLacks
+  ) where
+
+-- Must not be exported
+foreign import data RowLacking :: Symbol -> Symbol -> Type
+
+class RowLacksF (key :: Symbol)
+                (row :: # Type)
+                (typ :: Type)
+
+instance rowLacksF
+  :: ( RowCons key typ () singleton
+     , Union row singleton composed
+     , RowCons key typ rest composed )
+  => RowLacksF key row typ
+
+-- | Encodes the constraint that a given row does not contain a
+-- | specific key.
+class RowLacks (key :: Symbol)
+               (row :: # Type)
+
+instance rowLacks
+  :: ( RowLacksF key row (RowLacking "Existing key found in row" key)
+     , RowLacksF key row (RowLacking "Existing key found polymorphically in row" key) )
```

Comment:
The first constraint makes sense to me, but I don't understand why this second constraint is needed.